### PR TITLE
Fix many errors found by PVS-Studio

### PR DESCRIPTION
**Pull request by aaronfranke**
Fix errors 2, 3, 4, 6, 8, 9, 11, 12, 13, 14, and 15. See https://github.com/godotengine/godot/issues/24014

### CODE REVIEW

File: editor/editor_spin_slider.cpp [hunk 1]
```diff
@@ -154,7 +154,7 @@ void EditorSpinSlider::_grabber_gui_input(const Ref<InputEvent> &p_event) {
 
 void EditorSpinSlider::_notification(int p_what) {
 
-	if (p_what == MainLoop::NOTIFICATION_WM_FOCUS_OUT || p_what == MainLoop::NOTIFICATION_WM_FOCUS_OUT) {
```
Comment (akien-mga) [inline editor/editor_spin_slider.cpp]: Maybe it was meant to be `NOTIFICATION_WM_FOCUS_IN`? I doubt reduz would have gone through the pain of a copy paste and `||` condition if it wasn't intended.

File: scene/resources/texture.cpp [hunk 2]
```diff
@@ -1084,11 +1084,10 @@ bool AtlasTexture::get_rect_region(const Rect2 &p_rect, const Rect2 &p_src_rect,
 
 bool AtlasTexture::is_pixel_opaque(int p_x, int p_y) const {
 
-	if (atlas.is_valid()) {
-		return atlas->is_pixel_opaque(p_x + region.position.x + margin.position.x, p_x + region.position.y + margin.position.y);
-	}
+	int x = p_x + region.position.x + margin.position.x;
```
Comment (akien-mga) [inline scene/resources/texture.cpp]: Please keep the `if`, no need to compute `x` and `y` if the atlas is invalid.

**Comment by akien-mga** [comment]
Thanks!

**Comment by girng** [comment]
|| is not the same as && though, won't this cause issues in the future possibly?

**Comment by aaronfranke** [comment]
@girng The point is that it was causing issues before when it was `||`. It clearly looks like it was designed to check if the index was within a range, therefore `&&` is the correct operator.

**Comment by girng** [comment]
@aaronfranke aww, i see. within a range. okay was just checking lol ty

**Comment by akien-mga** [comment]
Cherry-picked for 3.0.7. (Only relevant file changes.)

**Pull request merged**